### ISSUE TITLE
Crash w/ 'linux-vdso.so.1': No such file

### ISSUE THREAD

**Issue by jkoudys** [CleverRaven/Cataclysm-DDA#66848]
### Describe the bug

crashes during utf8_to_wstr().

I'm running on debian bullseye, in a chroot under chromeos.

I've been playing for years and it was only after rebuilding maybe a month or so ago this started to happen.

It happens in all games all the time, it's not something that's popped up in this save file only.

I do get the 'no valid unicode locale' message on startup, but I've had that for years.

### Attach save file

n/a

### Steps to reproduce

Three scenarios cause the same crash:
 * Getting an NPC to join
 * `a`ctivating anything near a car (even just a frame with a forklift arm installed)
 * entering a vehicle (doesn't do it for wrecks, they seem to need a dashboard or controls)

### Expected behavior

Game doesn't crash.

### Versions and configuration

[debug.log](https://github.com/CleverRaven/Cataclysm-DDA/files/12028727/debug.log)

### Additional context

[crash.log](https://github.com/CleverRaven/Cataclysm-DDA/files/12028730/crash.log)

**Comment by SurFlurer** [comment]
> I've been playing for years and it was only after rebuilding maybe a month or so ago this started to happen.

So you haven't rebuild in a month or so? What #65916 fixed sounds related. 

Unless the utf8 string is already "bad" from the start before translation, #65916 should have fixed this sort of crash.

**Comment by jkoudys** [comment]
No I've rebuilt several times since then, hoping it would fix it. Last build was July 5th.

Another issue is that all the names started showing as "Tom" around this time, which as I understand it is what happens when a name fails to load. NPC names, world names, my name, town names.. all Tom. It's like MySpace in this cataclysm.

I went in and did:
```
apt-get install -y locales
```
like you'd suggested and set my lang to my region's locale (en_CA.utf8). The warning at the start doesn't show, and now I can go in cars!

I think the action for my issue should be to at very least make the warning at the start firmer. It reads like you can play, it'll just look a bit garbled in some places. In practice the game crashes and has odd behaviour. Unless you REALLY want to support full ascii mode like this is Nethack or something. It might also make sense to not start at all. Better it gives you an error with some instructions on how to resolve it, than to let you play then start having problems.

**Comment by Qrox** [comment]
> 'linux-vdso.so.1': No such file

According to your crash log, that message was from addr2line which failed to locate the file when trying to load debug symbol after the crash, and was not related to the crash otherwise. It crashed when trying to converting strings to upper case when displaying vehicle items.

> Another issue is that all the names started showing as "Tom" around this time

The Tom bug was fixed by #66460.

> It reads like you can play, it'll just look a bit garbled in some places.

That is true for some environments (such as MSYS2 MSYS) at least, so locking the player out of the game because they don't have a locale is not an option. Making the warnings firmer as you suggested is something that can be done though, if it turns out to be related to the crash.
